message_ix/core.py

```diff
@@ -107,3 +116,73 @@ def vintage_and_active_years(self):
         year_pairs = [(y_v, y_a) for y_v, y_a in combinations if y_v <= y_a]
         v_years, a_years = zip(*year_pairs)
         return v_years, a_years
+
+    def solve(self, **kwargs):
+        """Solve a Message Scenario. See ixmp.Scenario.solve() for arguments"""
+        return super(Scenario, self).solve(model='MESSAGE', **kwargs)
+
+    def clone(self, model=None, scen=None, annotation=None, keep_sol=True,
+              first_model_year=None):
+        """clone the current scenario and return the new scenario
+
+        Parameters
+        ----------
+        model : string
+            new model name
+        scen : string
+            new scenario name
+        annotation : string
+            explanatory comment (optional)
+        keep_sol : boolean, default: True
+            indicator whether to include an existing solution
+            in the cloned scenario
+        first_model_year: int, default None
+            new first model year in cloned scenario
+            ('slicing', only available for MESSAGE-scheme scenarios)
+        """
+        self._keep_sol = keep_sol
+        self._first_model_year = first_model_year or 0
+        model = self.model if not model else model
+        scen = self.scenario if not scen else scen
+        return Scenario(self.platform, model, scen, annotation=annotation,
+                        cache=self._cache, clone=self)
+
+    def rename_technology(self, from_tech, to_tech):
```

Comment:
Suggest to refactor to `rename(set, mapping)`, where `set` is the name of the set in which you want to do the renaming and `mapping` is a dictionary...